File: app/src/main/java/rubik/shifttest/data/repository/UserRepositoryImpl.java
package rubik.shifttest.data.repository;

import rubik.shifttest.data.storage.models.User;
import rubik.shifttest.data.storage.UserStorage;
import rubik.shifttest.domain.models.UserRegisterCredential;
import rubik.shifttest.domain.repository.UserRepository;

public class UserRepositoryImpl implements UserRepository {

    private UserStorage userStorage;

    public UserRepositoryImpl(UserStorage userStorage) {
        this.userStorage = userStorage;
    }

    @Override
    public boolean saveUserRegisterCredential(UserRegisterCredential userRegisterCredential) {
        return userStorage.saveUser(mapToData(userRegisterCredential));
    }

    /**
     * @return null if shared preferences with $KEY_USER_REGISTER_CREDENTIAL is not exist
     *          or an UserRegisterCredential object
     */
    @Override
    public UserRegisterCredential getUserRegisterCredential() {
        return mapToDomain(userStorage.getUser());
    }

    private UserRegisterCredential mapToDomain(User user) {
        return new UserRegisterCredential(user.getFirstName(), user.getLastName(), user.getBornDate(), user.getPassword());
    }

    private User mapToData(UserRegisterCredential userRegisterCredential) {
        return new User(userRegisterCredential.getFirstName(),
                userRegisterCredential.getLastName(), userRegisterCredential.getBornDate(), userRegisterCredential.getPassword());
    }
}

File: shared/user/src/main/kotlin/ru/rubik/user/data/UserRepositoryImpl.kt
package ru.rubik.user.data

import ru.rubik.user.data.storage.UserStorage
import ru.rubik.user.data.storage.models.UserDTO
import ru.rubik.user.domain.entities.User
import ru.rubik.user.domain.repository.UserRepository

class UserRepositoryImpl(private val userStorage: UserStorage) : UserRepository {

	override fun saveUser(user: User) = userStorage.saveUser(mapToData(user))

	override fun getUser(): User = mapToDomain(userStorage.getUser())

	private fun mapToDomain(user: UserDTO): User =
		User(
			firstName = user.firstName,
			lastName = user.lastName,
			bornDate = user.bornDate,
			password = user.password,
		)

	private fun mapToData(user: User): UserDTO =
		UserDTO(
			firstName = user.firstName,
			lastName = user.lastName,
			bornDate = user.bornDate,
			password = user.password,
		)
}